flink-ml-iteration/src/main/java/org/apache/flink/iteration/utils/ReflectionUtils.java

```diff
@@ -80,4 +83,55 @@ public static <T> T callMethod(
                     "Failed to get method" + methodName + " from " + targetObject, e);
         }
     }
+
+    /**
+     * The utility method to call method with the specific name and parameters.
+     *
+     * <p>Note that this method is added only for bypassing the existing bug in Py4j. It doesn't
+     * validate the classes of parameters so it can only deal with the classes that have only one
+     * method with the specific name.
+     *
+     * <p>TODO Remove this method after the Py4j bug is fixed.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> T callMethodByName(
+            Object targetObject, Class<?> declaredClass, String methodName, Object[] parameters) {
+        List<Method> methods = new ArrayList<>();
+        for (Method m : declaredClass.getMethods()) {
+            if (methodName.equals(m.getName())) {
+                m.setAccessible(true);
+                methods.add(m);
+            }
+        }
+        Preconditions.checkState(
+                methods.size() == 1,
+                "Only one method with name %s is permitted to be declared in %s",
+                methodName,
+                declaredClass);
+        try {
+            return (T) methods.get(0).invoke(targetObject, parameters);
+        } catch (Exception e) {
+            throw new RuntimeException(
+                    "Failed to invoke method" + methodName + " from " + targetObject, e);
+        }
+    }
+
+    /**
+     * The utility method to call method with the specific name.
+     *
+     * <p>TODO Remove this method after the Py4j bug is fixed.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> T callMethodByName(
```

Comment:
Would it be simpler to update the existing `callMethod(...)` to also use the inherited methods?

Reply:
Would it be better if we keep the logic that is to be removed in a separate method? We don't have the need to use the inherited methods except for Py4j, right? So we don't have to introduce this change in the existing method.

Based on this reason, I add the method called `callMethodByName(Object targetObject, Class<?> declaredClass, String methodName)` to differ from the existing `callMethod(Object targetObject, Class<?> declaredClass, String methodName)` which has the same parameters. To keep the same naming style in the two methods I added, they are all named with `callMethodByName`, and this answers your above question.